### PR TITLE
[2.6] Add January k8s patches

### PR DESCRIPTION
**Pull request by doflamingo721**
Linked issue:
https://github.com/rancher/rancher/issues/36029

Adds:
1.20.15-rancher1-1
1.21.9-rancher1-1
1.22.6-rancher1-1

### CODE REVIEW

File: rke/k8s_rke_system_images.go [hunk 1]
```diff
@@ -6263,9 +6263,9 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			Nodelocal:                 "rancher/mirrored-k8s-dns-node-cache:1.15.13",
 		},
 		// Enabled out of band post v2.6.3
-		"v1.20.14-rancher2-1": {
+		"v1.20.15-rancher1-1": {
```
Comment (a-blender) [inline rke/k8s_rke_system_images.go:6310]: @kinarashah Does this mean that the kubernetes versions in `templates.go` will have to be bumped as well? Coredns restricts for 1.20.14 and 1.21.8

File: rke/k8s_rke_system_images.go [hunk 2]
```diff
@@ -6571,9 +6571,9 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			Nodelocal:                 "rancher/mirrored-k8s-dns-node-cache:1.18.0",
 		},
 		// Enabled out of band post v2.6.3
-		"v1.21.8-rancher2-1": {
+		"v1.21.9-rancher1-1": {
```
Comment (a-blender) [inline rke/k8s_rke_system_images.go:6662]: @kinarashah Same question

File: rke/k8s_rke_system_images.go (hunk 1 above)
Comment (kinarashah) [reply rke/k8s_rke_system_images.go:6310]: @annablender You're right, tho in this case the template conditions are `>=1.20.14` so it'll cover new versions moving forward.

**Review by kinarashah — CHANGES_REQUESTED** [review]
The versions that are getting updated here are already released (merged to release-v2.6), so we need to add new entries (instead of updating `rancher2-1`). 

If you rebase your PR, the build should catch this and fail.

**Comment by doflamingo721** [comment]
@kinarashah I rebased my PR and added new entries as you mentioned. Can you please review it again?

File: data/data.json [hunk 3]
```diff
@@ -8653,6 +8653,49 @@
    "aciGbpServerContainer": "noiro/gbp-server:5.1.1.0.1ae238a",
    "aciOpflexServerContainer": "noiro/opflex-server:5.1.1.0.1ae238a"
   },
+  "v1.20.15-rancher1-1": {
+   "etcd": "rancher/mirrored-coreos-etcd:v3.4.15-rancher1",
+   "alpine": "rancher/rke-tools:v0.1.78",
```
Comment (kinarashah) [inline data/data.json:8658]: we need to bump rke-tools to v0.1.79 for all versions

Comment (kinarashah) [reply data/data.json:8658]: @doflamingo721 could you please also update them here https://github.com/rancher/kontainer-driver-metadata/pull/780 so both PRs would be ready to merge

**Review by kinarashah — APPROVED** [review]

**Pull request merged**